Gems/GradientSignal/Code/Source/Editor/EditorGradientBakerComponent.cpp

```diff
@@ -155,6 +209,181 @@ namespace GradientSignal
+        // so need to use RGBA format for them
+        int channels = 1;
+        if (fullPathIO.Extension() == ".tga" || fullPathIO.Extension() == ".exr")
+        {
+            channels = 4;
+        }
+
+        int bytesPerPixel = 0;
+        OIIO::TypeDesc pixelFormat = OIIO::TypeDesc::UINT8;
+        switch (m_configuration.m_outputFormat)
+        {
+        case OutputFormat::R8:
+            bytesPerPixel = 1;
+            pixelFormat = OIIO::TypeDesc::UINT8;
+            break;
+        case OutputFormat::R16:
+            bytesPerPixel = 2;
+            pixelFormat = OIIO::TypeDesc::UINT16;
+            break;
+        case OutputFormat::R32:
+            bytesPerPixel = 4;
+            pixelFormat = OIIO::TypeDesc::FLOAT;
+            break;
+        }
+        const size_t imageSize = imageResolutionX * imageResolutionY * channels * bytesPerPixel;
+        AZStd::vector<AZ::u8> pixels;
+        pixels.resize(imageSize, 0);
+
+        AZ::IO::Path absolutePath = fullPathIO.LexicallyNormal();
+        std::unique_ptr<OIIO::ImageOutput> outputImage = OIIO::ImageOutput::create(absolutePath.c_str());
+        if (!outputImage)
+        {
+            AZ_Error("GradientBaker", false, "Failed to write out gradient baked image to path: %s",
+                absolutePath.c_str());
+            return;
+        }
+
+        OIIO::ImageSpec spec(imageResolutionX, imageResolutionY, channels, pixelFormat);
+        outputImage->open(absolutePath.c_str(), spec);
+
+        AZ::Aabb inputBounds = GetPreviewBounds();
+        AZ::EntityId boundsEntityId = GetPreviewEntity();
+
+        const AZ::Vector3 inputBoundsCenter = inputBounds.GetCenter();
+        const AZ::Vector3 inputBoundsExtentsOld = inputBounds.GetExtents();
+        inputBounds =
+            AZ::Aabb::CreateCenterRadius(inputBoundsCenter, AZ::GetMax(inputBoundsExtentsOld.GetX(), inputBoundsExtentsOld.GetY()) / 2.0f);
+
+        const AZ::Vector3 inputBoundsStart =
+            AZ::Vector3(inputBounds.GetMin().GetX(), inputBounds.GetMin().GetY(), inputBoundsCenter.GetZ());
+        const AZ::Vector3 inputBoundsExtents = inputBounds.GetExtents();
+        const float inputBoundsExtentsX = inputBoundsExtents.GetX();
+        const float inputBoundsExtentsY = inputBoundsExtents.GetY();
+
+        // Get the largest square size that fits into our window bounds.
+        const int imageBoundsX = AZStd::min(imageResolutionX, imageResolutionY);
+        const int imageBoundsY = AZStd::min(imageResolutionX, imageResolutionY);
+
+        // Get how many pixels we need to offset in x and y to center our square in the window.  Because we've made our square
+        // as large as possible, one of these two values should always be 0.
+        // i.e. we'll end up with black bars on the sides or on top, but it should never be both.
+        const int centeringOffsetX = (imageResolutionX - imageBoundsX) / 2;
+        const int centeringOffsetY = (imageResolutionY - imageBoundsY) / 2;
```

Comment:
It seems like we should either completely fill the requested image size, or we shouldn't allow them to choose non-square images. It seems a bit wasteful to allow non-square images that get padded with black pixels.

Reply:
Spoke offline with Mike, will update this logic to respect the requested image size instead of forcing to a square image.